### ISSUE TITLE
Dist folder is missing

### ISSUE THREAD

**Issue by anthonny** [asciidoctor/asciidoctor.js#172]
The dist folder is not in the repository anymore, so i can't use asciidoctor 1.5.4 in a backend project : 

`[13:32:50] gulp-notify: [Compile Error] Cannot find module 'asciidoctor.js' from '/Users/anthonny/github/hubpress.io/src/hubpress/scripts/utils'`

From the `package.json` : 

`"main": "dist/npm/asciidoctor-core.min.js",`

**Comment by Mogztter** [comment]
I think I misunderstood what `npm publish` does. This command doesn't publish the latest tag but the working directory :disappointed:

**Comment by Mogztter** [comment]
@mojavelinux We will need to publish a new version. Unfortunately `1.5.4-1` has a lower precedence than the associated normal version in SemVer. So we will need to publish a `1.5.5` (something) version: `1.5.5-beta.1` ?

Anthonny alos raises a good point about #73. Now that we removed dist folder from master, it's not possible anymore to build a Node module against Asciidoctor.js master because the dist folder is missing. With previous version Anthonny could add a dependency on `asciidoctor.js` master from `package.json`:


```json
"dependencies": {
    "asciidoctor.js": "asciidoctor/asciidoctor.js"
}
```

**Comment by mojavelinux** [comment]
> I think I misunderstood what npm publish does. This command doesn't publish the latest tag but the working directory

I hit exactly the same problem once when publishing a bespoke plugin. That behavior was not at all what I expected.

If 1.5.4-1 has lower precedence than 1.5.4, then 1.5.5-1 will have a lower precedence then 1.5.5. Perhaps we can use that to our advantage here. Yes? (But I'd also be okay with 1.5.5-beta.1).

> Now that we removed dist folder from master, it's not possible anymore to build a Node module against Asciidoctor.js master because the dist folder is missing.

I'm going to hold firm on this because I consider to it be a really bad practice to be publishing dist into master. Putting them there also breaks the entire development process because every PR ends up with all the dist files. It's just not good software engineering, IMO.

I'd be okay with an integration branch. 'integration' (or 'latest') would have the dist folder from a nightly build and 'master' is where the code gets worked on. npm accepts any git url, which can be a branch, as I understand it.

**Comment by Mogztter** [comment]
Ok, I've just released `1.5.5-1`, sorry about that :disappointed: 

> I'd be okay with an integration branch. 'integration' (or 'latest') would have the dist folder from a nightly build and 'master' is where the code gets worked on. npm accepts any git url, which can be a branch, as I understand it.

That's a good idea, I will see what I can do!

**Comment by mojavelinux** [comment]
+1

You didn't make any mistake. We simply learned something we didn't know :)

**Comment by lance** [comment]
Looks like a version of 1.5.5-1 won't help those folks who have loose semver dependencies in their package.json. The `-1` suffix is considered a pre-release suffix, and so 1.5.5 is still > 1.5.5-1. See the issue I opened on the metalsmith-asciidoctor project. https://github.com/ndhoule/metalsmith-asciidoc/issues/2

**Comment by Mogztter** [comment]
> The -1 suffix is considered a pre-release suffix, and so 1.5.5 is still > 1.5.5-1.

You mean 1.5.4 is still > 1.5.5-1 ? The previous release was 1.5.4.

**Comment by lance** [comment]
Yes, my mistake.

Lance Ball
http://lanceball.com

**Comment by Mogztter** [comment]
This issue is resolved with the `1.5.5-1` release: https://github.com/asciidoctor/asciidoctor.js/releases/tag/v1.5.5-1
